src/components/lobby/MatchesList.js

```diff
@@ -78,17 +83,43 @@ const matchesListItemPropTypes = {
 };
 
 const MatchesListItem = React.memo(
-  ({ match, match: { matchID, players, setupData }, game, onJoin, current, disabled }) => {
-    const { t } = useTranslation("lobby");
+  ({
+    match,
+    match: { matchID, gameName, players, setupData },
+    game,
+    onJoin,
+    current,
+    disabled,
+  }) => {
+    const { t } = useTranslation("lobby", "translation");
+    const user = useUser();
+    const credentials = useCredentials();
+    const gameCredentials = credentials && credentials[matchID];
+    const history = useHistory();
+    const currentPlayers = players.filter((p) => p.name);
+
+    const handleLeave = useCallback(() => {
```

Comment:
That's ugly - pretty much the same code is in `LeaveButton` component. We should make it more generic and reuse in both places 🤔

Reply:
There is most likely a better way to do this, but that's basically as good as my js skills get 🙈